src/geo/transform.js

```diff
@@ -569,10 +569,14 @@ class Transform {
         // No change of LOD behavior for pitch lower than 60 and when there is no top padding: return only tile ids from the requested zoom level
         const minZoom = this.pitch <= 60.0 && this._edgeInsets.top <= this._edgeInsets.bottom && !this._elevation ? z : 0;
 
-        const maxRange = this.elevation ? this.elevation.exaggeration() * 10000 : 0;
+        // When calculating DEM tile cover, create deep AABB for nodes, to ensure they intersect frustum.  For other sources,
+        // use 2D tile cover: tile assumes center elevation until corresponding DEM cover is loaded to tell if additional
+        // tiles need to be loaded.
+        const maxRange = options.isTerrainDEM && this._elevation ? this._elevation.exaggeration() * 10000 : this._centerAltitude;
+        const minRange = options.isTerrainDEM && this._elevation ? -maxRange : this._centerAltitude;
```

Comment:
Have you considered using the min and max altitude of DEMs in view as the range instead of this._centerAltitude, or is this choice arbitrary and doesn't matter?

Reply:
I have.
Often in mountain areas near cost, and especially if the pitch is large (due to LOD) there could be sea level contribution in one tile leading to loading unnecessary tiles.
Initial patch here was done so that tiles with no DEM information were skipped from returned cover. In a way, if we're not sure the tile is in cover and what the tile zoom is, then it is not in cover... forcing DEM tile to load first before making request for other tiles.
The problem there was that when transitioning from 2D->3D, we would see background only (or skybox, depending on implementation) until DEM tiles would get loaded and this wasn't looking good.

The choice matters: this._centerAltitude is in most of the cases, e.g. elevated area with street level zoom, the best choice to reuse 2D cover when turning on terrain and also closest to expected zoom level in 3D.